lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1808,9 +1808,11 @@ namespace Js
                 Output::Print(_u("MissingPropertyCaching: Missing property %d on slow path.\n"), propertyId);
             }
 
-            // Only cache missing property lookups for non-root field loads on objects that have PathTypeHandlers, because only these objects guarantee a type change when the property is added,
-            // which obviates the need to explicitly invalidate missing property inline caches.
-            if (!PHASE_OFF1(MissingPropertyCachePhase) && !isRoot && DynamicObject::Is(instance) && ((DynamicObject*)instance)->GetDynamicType()->GetTypeHandler()->IsPathTypeHandler())
+            // Only cache missing property lookups for non-root field loads on objects that have PathTypeHandlers and DictionaryTypeHandlers, because only these types have the right behavior
```

Comment:
can't we invalidate this for all types by default?

pros -> use this approach for all + don't do type checking.
cons -> ??

Reply:
We could, and I think the only costs would be the implementation cost of adding the prototype-cache invalidation logic to the other kinds of types, and any potential performance costs that arise because we invalidate caches more often than previously.  IB and benchmark results, however, indicate that this latter cost is likely to be negligible.

Reply:
copy paste from your comments below;

```
However, if the new property is already in the cache as a missing property, we have to invalidate the prototype caches.
```
Just asking;
How this might affect the perf badly? It sounds like we better do it already for others too?

Reply:
If we are able to say that the only cost here is to implement this for others.. well, could be nice to at least have an open `performance` issue on this. Curious, how that could affect Acme.

Reply:
It might affect perf because we're invalidating the caches more frequently than previously; this could in principle cause operations that currently hit the cache to start missing.  As above, though, we don't expect this cost to be significant.

I assume that we're already invalidating prototype caches correctly for other type handlers (that is, under the same conditions in which we invalidate the prototype caches for dictionary type handlers in the code currently on master), but I haven't verified this explicitly.  This should be sufficient as long as we don't allow the other type handlers into missing-property caches.